airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -473,60 +490,116 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
+
+      final List<AutoCloseableIterator<AirbyteMessage>> ctidIterators = new ArrayList<>(ctidHandler.getIncrementalIterators(
+          new ConfiguredAirbyteCatalog().withStreams(finalListOfStreamsToBeSyncedViaCtid), tableNameToTable, emittedAt));
+      final List<AutoCloseableIterator<AirbyteMessage>> xminIterators = new ArrayList<>(xminHandler.getIncrementalIterators(
+          new ConfiguredAirbyteCatalog().withStreams(streamsCategorised.remainingStreams().streamsForXminSync()), tableNameToTable, emittedAt));
+
       return Stream
-          .of(ctidIterator, xminIterator)
+          .of(ctidIterators, xminIterators)
+          .flatMap(Collection::stream)
+          .collect(Collectors.toList());
+
+    } else if (isIncrementalSyncMode(catalog) && ctidFeatureFlags.isCursorSyncEnabled()) {
+      final PostgresStandardStateManager postgresStandardStateManager = new PostgresStandardStateManager(stateManager.getRawStateMessages(), catalog);
+      final StreamsCategorised<StandardStreams> streamsCategorised = categoriseStreams(postgresStandardStateManager, catalog);
+      final List<AirbyteStreamNameNamespacePair> streamsUnderVacuum = streamsUnderVacuum(database,
+          streamsCategorised.ctidStreams().streamsForCtidSync(),
+          getQuoteString());
+      final List<ConfiguredAirbyteStream> finalListOfStreamsToBeSyncedViaCtid =
+          filterStreamsUnderVacuumForCtidSync(streamsUnderVacuum, streamsCategorised.ctidStreams());
+      final Map<io.airbyte.protocol.models.AirbyteStreamNameNamespacePair, Long> fileNodes =
+          PostgresQueryUtils.fileNodeForStreams(database,
+              finalListOfStreamsToBeSyncedViaCtid,
+              getQuoteString());
+      final CtidStateManager ctidStateManager = new CtidPerStreamStateManager(streamsCategorised.ctidStreams().statesFromCtidSync(), fileNodes);
+      final Map<io.airbyte.protocol.models.AirbyteStreamNameNamespacePair, TableBlockSize> tableBlockSizes =
+          PostgresQueryUtils.getTableBlockSizeForStream(
+              database,
+              finalListOfStreamsToBeSyncedViaCtid,
+              getQuoteString());
+      if (finalListOfStreamsToBeSyncedViaCtid.isEmpty()) {
+        LOGGER.info("No Streams will be synced via ctid.");
+      } else {
+        LOGGER.info("Streams to be synced via ctid : {}", finalListOfStreamsToBeSyncedViaCtid.size());
+      }
+
+      if (!streamsCategorised.remainingStreams().streamsForStandardSync().isEmpty()) {
+        LOGGER.info("Streams to be synced via standard cursor : {}", streamsCategorised.remainingStreams().streamsForStandardSync().size());
+      } else {
+        LOGGER.info("No streams to be synced via standard cursor");
+      }
+
+      final Map<io.airbyte.protocol.models.AirbyteStreamNameNamespacePair, StandardStatus> standardStatusMap =
+          getStandardSyncStatusForStreams(database, finalListOfStreamsToBeSyncedViaCtid, postgresStandardStateManager);
```

Comment:
nit : can we rename "standard" to "cursorBased" or something like that? I think it makes it clearer the type of incremental sync we're doing. WDYT?

Reply:
Here and other places in the PR

Reply:
@akashkulk, I'm good with changing it to cursorBased. I chose standard cause it made sense at the time, but no strong feelings towards it. Will make the change.